reader/views.py

```diff
@@ -125,7 +125,25 @@ def render_react_component(component, props):
     `props` may either be JSON (to save reencoding) or a dictionary.
     Returns HTML.
     """
-    if not USE_NODE or get_shared_cache_elem("regenerating"):
+    SHARED_CACHE_KEYS = [
+        "books_en_json",
+        "topic_toc_json",
+        "last_cached",
+        "term_mapping_json",
+        "toc_json",
+        "term_mapping",
+        "topic_toc_category_mapping",
+        "topic_toc",
+        "toc",
+        "books_en",
+        "virtualBooks",
+    ]
+
+    if not USE_NODE or not check_shared_multiple_keys(SHARED_CACHE_KEYS):
```

Comment:
We don't want to be regenerating the shared cache just because USE_NODE is false.